subprojects/build-cache-http/src/integTest/groovy/org/gradle/caching/http/internal/HttpBuildCacheServiceTest.groovy

```diff
@@ -94,15 +94,15 @@ class HttpBuildCacheServiceTest extends Specification {
     }
 
     def "can cache artifact with redirect"() {
-        def destFile = tempDir.file("cached.zip")
         def content = "Data".bytes
         server.expectPutRedirected("/cache/${key.hashCode}", "/redirect/cache/${key.hashCode}")
-        server.expectPut("/redirect/cache/${key.hashCode}", destFile, HttpStatus.SC_OK, null, content.length)
 
         when:
         cache.store(key, writer(content))
         then:
-        destFile.bytes == content
+        BuildCacheException exception = thrown()
+
+        exception.message == "Storing entry at '${server.uri}/cache/${key.hashCode}' response status 302: Found"
```

Comment:
We should print an error specific to redirect errors explaining what's wrong. We do understand the situation, so why not help users?

Reply:
Now we print a descriptive error.